### ISSUE TITLE
Standardize on base64 encoded, SHA-256 hashed, normalized URIs as Redis keys

### ISSUE THREAD

**Issue by humphd** [Seneca-CDOT/telescope#522]
**What would you like to be added**:

Our `Feed` (see #520) and `Post` objects need to live in Redis.  I've been doing a bit of research on the best way to do this, and here is what I suggest:

1. A `Feed` is a Redis Hash stored using the feed's URL as its key.  Let's use [normalize-url](https://www.npmjs.com/package/normalize-url) to clean it up a bit (so we don't have slight variations leading to duplication) and then hash it using MD5 and base64 encode it (see below).  We should also be namespacing our keys.  I suggest `t:feed:<base64 encoded MD5 hash>`, where `t=Telescope`.
2. A `Post` is a Redis Hash stored using the article's `guid`, which is (likely) a URI or URL.  Let's hash using MD5 and then base64 encode it.  I suggest `t:post:<base64 encoded MD5 hash>`.

To hash and encode these URIs in node, do this:

```
const crypto = require('crypto')

function toBase64(url) {
  return crypto.createHash('md5').update('some_string').digest('base64');
}
```

We'd also need a way to compare a URL with a key, so we can check if a URI is already in the database.

**Why would you like this to be added**:

I noticed [this post](https://groups.google.com/forum/m/#!topic/redis-db/dPW3m9t6vk4) by Redis' creator suggesting this approach for storing URLs.

**Comment by humphd** [comment]
Reading https://twitter.com/joepie91/status/1218867444489641985 made me think that we might want something other than MD5 here.  We just need "unique" without security consideration, where fast is important.  I'm not sure what the best algorithm would be.  On my Mac, openssl has these implemented:

```
$ openssl list -digest-algorithm
openssl:Error: 'list' is an invalid command.

Standard commands
asn1parse         ca                certhash          ciphers
crl               crl2pkcs7         dgst              dh
dhparam           dsa               dsaparam          ec
ecparam           enc               errstr            gendh
gendsa            genpkey           genrsa            nseq
ocsp              passwd            pkcs12            pkcs7
pkcs8             pkey              pkeyparam         pkeyutl
prime             rand              req               rsa
rsautl            s_client          s_server          s_time
sess_id           smime             speed             spkac
ts                verify            version           x509

Message Digest commands (see the `dgst' command for more details)
gost-mac          md4               md5               md_gost94
ripemd160         sha1              sha224            sha256
sha384            sha512            streebog256       streebog512
whirlpool

Cipher commands (see the `enc' command for more details)
aes-128-cbc       aes-128-ecb       aes-192-cbc       aes-192-ecb
aes-256-cbc       aes-256-ecb       base64            bf
bf-cbc            bf-cfb            bf-ecb            bf-ofb
camellia-128-cbc  camellia-128-ecb  camellia-192-cbc  camellia-192-ecb
camellia-256-cbc  camellia-256-ecb  cast              cast-cbc
cast5-cbc         cast5-cfb         cast5-ecb         cast5-ofb
chacha            des               des-cbc           des-cfb
des-ecb           des-ede           des-ede-cbc       des-ede-cfb
des-ede-ofb       des-ede3          des-ede3-cbc      des-ede3-cfb
des-ede3-ofb      des-ofb           des3              desx
rc2               rc2-40-cbc        rc2-64-cbc        rc2-cbc
rc2-cfb           rc2-ecb           rc2-ofb           rc4
rc4-40
```

**Comment by humphd** [comment]
Probably we can just swap out MD5 for SHA-256:

```js
crypto.createHash('sha-256').update('some-string').digest('base64')
```

**Comment by humphd** [comment]
One bug this will fix, that our current implementation has, is that we won't keep duplicating feeds in Redis.  The fix in #547 really exposes it, since it keeps doubling, tripling, etc. the feeds that have to get processed.  So we should probably do this sooner vs. later.

**Comment by manekenpix** [comment]
I'll try to have a PR by the end of today or tomorrow morning.
